JonnyHicks-Portfolio

KidFromCalifornia opened the pull request:
Figma: (https://www.figma.com/design/TRJIJC4vbVQ3SYhvTibPCq/Jonny-HIcks-Portfolio-Design?node-id=1-350&m=dev)

https://jonnyhicks.netlify.app
or http://jonnyhicks.se/ (waiting for verification)

KidFromCalifornia committed:
CSS stying started  and link fonts and icons

KidFromCalifornia committed:
added assets

KidFromCalifornia committed:
Add initial component files for buttons and portrait

- Created ButtonLinks.jsx, Lightbuttons.jsx, and OrangeButtons.jsx as new components.
- Added Portrait.jsx component to display a portrait image.
- Introduced Portrait.css for styling the portrait component.
- Created cards.jsx as a placeholder for future card components.

KidFromCalifornia committed:
Refactor AboutMe and Portrait components; remove unused files and add initial styles

KidFromCalifornia committed:
Refactor App component; import AboutMe and remove unused files

KidFromCalifornia committed:
reanmed folders

KidFromCalifornia committed:
Add FontAwesome icons, refactor AboutMe component, and implement ButtonLinks;
added basic stlying

KidFromCalifornia committed:
Navbar Added

KidFromCalifornia committed:
Refactor ButtonLinks component; adjust import statement and clean up Navbar styles

KidFromCalifornia committed:
Refactor App and Card components; implement AboutMe section and update styles

KidFromCalifornia committed:
Refactor project structure; remove unused components and styles, update AboutMe and Projects sections

KidFromCalifornia committed:
Add Toolbox section, refactor App and Card components, update AboutMe and Projects sections

KidFromCalifornia committed:
Refactor button styles in GithubButton and NetlifyButton; update Projects component to use default project data

KidFromCalifornia committed:
jsons files up to date with dtat for  cards.

KidFromCalifornia committed:
Refactor project structure: update React and ReactDOM versions, enhance App component with new sections (Skills, Updates, Footer), and improve styling for buttons and navigation. Add Contacts and FooterBackground components, and implement Updates section with dynamic data. Remove unused MyJournal.json and update projects.json for consistency.

KidFromCalifornia committed:
Refactor project structure: reorganize sections and components, enhance styling for AboutMe, Projects, and Footer sections, and implement responsive design. Remove unused files and update CSS for better layout and alignment.

KidFromCalifornia committed:
Add Updates section with dynamic data, refactor UpdateCard and PostButton components, and enhance styling for better layout

KidFromCalifornia committed:
Refactor project structure and update component imports

- Updated import paths for Header and Updates components in App.jsx
- Deleted unused ButtonLinks, Card, and UpdateCard components along with their styles
- Refactored ContactDetails to use the new ButtonLinks path
- Simplified FooterBackground text
- Removed old styles and components related to buttons and cards
- Created new ButtonLinks component with styles
- Reorganized Navbar component and styles
- Updated AboutMe and Projects sections to reflect new component paths
- Added new styles for updates and refactored Updates section

KidFromCalifornia committed:
Refactor components and styles: remove unused LinkButton, update NetlifyButton and PostButton styles, enhance Card and UpdateCard layouts, and implement Toolbox section with new components.

KidFromCalifornia committed:
Refactor project structure: remove unused components (Backend, Creative, Frontend, Workflow), update button styles and classes, enhance Card component with Font Awesome icons, and improve layout for various sections.

KidFromCalifornia committed:
Update dependencies and enhance Card component styles with Font Awesome icons

KidFromCalifornia committed:
Refactor components: remove unused buttons and styles, implement OpenLink component, and enhance Card and UpdateCard layouts with styled-components.

KidFromCalifornia committed:
Refactor code structure for improved readability and maintainability

KidFromCalifornia committed:
Refactor OpenLink component to use styled-components, enhance Portrait layout with new shadow effect, and improve AboutMe section styles.

KidFromCalifornia committed:
Refactor button components: remove unused CSS files, enhance OpenLink component with styled-components, and improve layout in Projects and Skills sections.

KidFromCalifornia committed:
Refactor project structure: rename Toolbox to Toolkit, remove unused tech list components, and enhance styling with new SectionContainer and CardContainer components.

KidFromCalifornia committed:
Refactor components: enhance styling in ContactDetails and FooterBackground, replace SectionStyling with CardSectionStyling, and introduce ListsSectionStyles for improved layout in Skills and Toolkit sections.

KidFromCalifornia committed:
Refactor styling and layout: update OpenLink component to use styled-components, enhance CardSectionStyling and ListsSectionStyles with improved padding and margins, and adjust AboutMe section styles for better alignment and readability.

KidFromCalifornia committed:
Refactor layout and styling: enhance CardSectionStyling with margin, update CardStyles border color, improve FooterBackground responsiveness, remove Navbar component, adjust Portrait scaling, refine AboutMe section styles, and clean up Projects section rendering.

KidFromCalifornia committed:
Refactor HTML and JSX structure: update index.html with meta tags for SEO, rename Toolkit to ToolkitSection in App.jsx, enhance CardSectionStyling and FooterBackground components, improve layout in ContactDetails and ListsSectionStyles, fix typos in Techlists, and adjust Footer and Header styles for better responsiveness.

KidFromCalifornia committed:
Refactor components and styles: enhance layout and responsiveness in CardSectionStyling, ContactDetails, and ListsSectionStyles; remove FooterBackground component; adjust styles in Footer, Navbar, AboutMe, and Portrait; update CSS variables for improved color scheme and typography.

KidFromCalifornia committed:
testing ScrollAnimation class for scroll-triggered animations

KidFromCalifornia committed:
Refactor layout and styles: update footer and about me section for better responsiveness; enhance scroll animation with debugging logs; adjust list section alignment; improve CSS for consistent width across components.

KidFromCalifornia committed:
Refactor project structure and styles: remove unused components, update imports, and enhance styling for better organization and maintainability; add prop-types dependency for type checking.

KidFromCalifornia committed:
Refactor skills components: consolidate SkillsList into a single file, update imports in Skills and FooterContent components, and enhance styling for better organization.

KidFromCalifornia committed:
Refactor imports and styles: update import paths for components, remove unused CardStyles file, and enhance structure for better organization.

KidFromCalifornia committed:
Refactor Card and UpdateCard components: remove CardContainer, introduce CardGroup for better structure; enhance styling in ContactDetails and ProfileLinks; update Projects and Updates sections for improved layout and responsiveness.

KidFromCalifornia committed:
refactored styling on list sections

KidFromCalifornia committed:
Fix AboutMeContainer max-width to ensure proper
 layout on larger screens

KidFromCalifornia committed:
Implement code changes to enhance functionality and improve performance

KidFromCalifornia committed:
Fix og:image meta tag: change content attribute to href for proper linking

KidFromCalifornia committed:
testing og:image

KidFromCalifornia committed:
test

KidFromCalifornia committed:
fixed OGimage path

KidFromCalifornia committed:
Update og:image meta tag to correct image URL for better sharing visibility

KidFromCalifornia committed:
""

KidFromCalifornia committed:
refactor Meta tags

HIPPIEKICK reviewed (approved):
**A11y**
100% in Lighthouse ⭐

**Regarding the requirement: Your portfolio should follow the Figma design**
- Is it deliberate that your sections in the design are not aligned? I think it would look amazing if they at least were left-aligned. Same goes for between sections - if they all had the same spacing
- Your font-sizes are not according to the design
- I think it might become harder to maintain when you have a lot of different paddings and margins, e.g. on the Project section, the Card container has a margin and a padding around the whole thing + the Card group also has its own margin + the Section container also has its own margin and padding. I think you should be able to get rid of some of these. Also, maybe read up on the difference between padding and margin? 😇

**Codewise**
You have a good structure of your project and you’ve used components and props in a good way.

Great job keeping your entry point (App.jsx) clean. You could potentially move the <header> element to the Header component to clean it up even more.

A reminder about naming patterns, if it’s a component or folder specifically for one component it should be named with PascalCase. If it’s just a folder or helper JS file, it should be named in kebabCase. If it’s a folder for a specific component, the folder should be named the same as the component, e.g. AboutMe/AboutMe.

Idahel commented on src/App.jsx:
```diff
@@ -1,8 +1,27 @@
+      <header>
+        <Header />
+      </header>
+      <main>
```
I really like that you App.jsx is so clean!

Idahel reviewed (commented):
Hi Jonny! What a great job you did with your portfolio, it looks creative and personal! I like it!

I have looked through your code and could not find any specific improvements. I like that you have the data for your cards in json-files, which makes the jsx-files look cleaner. I also really like the fact that you use a nav bar, makes it easier to navigate. 

I only have two suggestions for "improvements", but it could also just be a matter of taste. The first is that I think the lists become a bit difficult to read when the text is centered. I have the same thing in the tablet view of my portfolio, but I'll probably change that. Especially when there's quite a bit of variation in how long the words are.

The second is that there were a lot of different files for a one-page website, and I had a little trouble navigating the code. But I also understand your choice given that the task was to practice using components.

That being said, I think you did a really great job, and I hope you're happy with the result!

KidFromCalifornia committed:
Refactor: Remove unused styled components and consolidate styles

- Deleted CardStyles, ListStyles, ListsSectionStyles, SvgShadow, and Techlists components as they were no longer needed.
- Moved relevant styles to new components and pages for better organization.
- Updated project and skills pages to utilize new component structure.
- Adjusted project icons and descriptions in projects.json for consistency.
- Improved footer and header
components for better layout and responsiveness.

converting to react router  currently broken

KidFromCalifornia committed:
Refactor: Update dependencies, improve routing, and enhance component imports

KidFromCalifornia committed:
Refactor: Clean up CSS imports, improve structure, and enhance accessibility features

KidFromCalifornia committed:
Refactor: Remove incomplete media query declaration

KidFromCalifornia committed:
Refactor: Rename Navbar import to Header for consistency

KidFromCalifornia committed:
Refactor: Rename Head component to Header and update import paths

KidFromCalifornia committed:
Fix: Reset Header.jsx filename case for Netlify

KidFromCalifornia committed:
Temp rename Header.jsx to header_temp.jsx

KidFromCalifornia committed:
Force add Header.jsx with correct case

KidFromCalifornia committed:
Clear git cache and re-add files to fix case issues
